### ISSUE TITLE
508-defect-2: VA national cemetery list loop needs design review (and possible re-design)

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#48287]
## Point of contact

Josh Kim

## Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

## Details

The VA national cemetery list loop page needs a design review as:
- it uses a typeahead with over 100+ cemeteries 
- it nests a list loop pattern within a dynamic expandable region (see below video on how recorded entries are erased when a user changes their decision)
- the list loop has heading issues and labelling issues for its buttons

https://user-images.githubusercontent.com/14154792/196208525-f599d5b5-09d1-4762-a297-7daa1b66b3e0.mov

## Reproduction steps

1. [Start the pre-need form as an unsigned in user](https://www.va.gov/burials-and-memorials/pre-need/form-10007-apply-for-eligibility/burial-benefits)
2. Select "other" for relationship to the service member
3. Reach the burial benefits section with the national cemetery selection

## Proposed solution or next steps

A designer should review this page and explore simpler, alternative methods for collecting this information like the [subtask pattern.](https://design.va.gov/patterns/forms/sub-task)

### References, articles, or WCAG support

1.
2.
3.
...

## Type of issue

- [X] Screenreader
- [ ] Keyboard
- [ ] Focus
- [X] Headings
- [ ] Color
- [ ] Zoom
- [X] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by sonntag-adam** [comment]
@ccruzgra @barron-charles

**Comment by ccruzgra** [comment]
This is the related issue logged for the differences in the Privacy Act Statement modal in staging v. prod: https://github.com/department-of-veterans-affairs/va.gov-team/issues/27577

**Comment by barron-charles** [comment]
Interesting, she must site-wide issue  just for our preneed application: https://github.com/department-of-veterans-affairs/va.gov-team/issues/27577#issuecomment-1021547655.

I see standard formatting for the privacy notice in staging for burial benefits: https://staging.va.gov/burials-and-memorials/application/530/introduction

Charlie Barron
UI/UX Designer
248-508-6794
***@***.******@***.***>

Booz I Allen I Hamilton
BoozAllen.com<http://BoozAllen.com>

From: Cindy Cruz ***@***.***>
Date: Tuesday, March 14, 2023 at 3:00 PM
To: department-of-veterans-affairs/va.gov-team ***@***.***>
Cc: Barron, Charles [USA] ***@***.***>, Mention ***@***.***>
Subject: [External] Re: [department-of-veterans-affairs/va.gov-team] 508-defect-2: VA national cemetery list loop needs design review (and possible re-design) (Issue #48287)

This is the related issue logged for the differences in the Privacy Act Statement modal in staging v. prod: #27577<https://urldefense.com/v3/__https:/github.com/department-of-veterans-affairs/va.gov-team/issues/27577__;!!May37g!IwNEGPRYvT_CZPKGMpAuW8xB2Kdff4_LiFHYmMjKALZdFO_GnlRdNrEO_aFxI_NVfm9tcJBdxiQYE0cDdrJJBdESQg$>

**Comment by barron-charles** [comment]
The last email missed a few points.


  1.  Interesting, she must see site-wide issues for the preneed application: https://github.com/department-of-veterans-affairs/va.gov-team/issues/27577#issuecomment-1021547655.


  1.  I see standard formatting for the privacy notice in staging for burial benefits here, so it’s just preneed : https://staging.va.gov/burials-and-memorials/application/530/introduction

Kind regards,
Charlie Barron
UI/UX Designer
248-508-6794
***@***.******@***.***>

Booz I Allen I Hamilton
BoozAllen.com<http://BoozAllen.com>

From: Cindy Cruz ***@***.***>
Date: Tuesday, March 14, 2023 at 3:00 PM
To: department-of-veterans-affairs/va.gov-team ***@***.***>
Cc: Barron, Charles [USA] ***@***.***>, Mention ***@***.***>
Subject: [External] Re: [department-of-veterans-affairs/va.gov-team] 508-defect-2: VA national cemetery list loop needs design review (and possible re-design) (Issue #48287)

This is the related issue logged for the differences in the Privacy Act Statement modal in staging v. prod: #27577<https://urldefense.com/v3/__https:/github.com/department-of-veterans-affairs/va.gov-team/issues/27577__;!!May37g!IwNEGPRYvT_CZPKGMpAuW8xB2Kdff4_LiFHYmMjKALZdFO_GnlRdNrEO_aFxI_NVfm9tcJBdxiQYE0cDdrJJBdESQg$>

**Comment by barron-charles** [comment]
@joshkimux We did some research and the erasing of data when they change options is a component issue that happens on most forms using a similar pattern. This was was seen in the [Rogers STEM Scholarship application](https://www.va.gov/education/other-va-education-benefits/stem-scholarship/apply-for-scholarship-form-22-10203/introduction?next=loginModal) and [Apply for VA Education Benefits](https://www.va.gov/education/apply-for-education-benefits/application/1990/introduction).

![Apply for VA Education Benefits](https://user-images.githubusercontent.com/95768999/225399205-966c076c-a69d-47c1-a2aa-f86af9943a8d.gif)

![Rogers STEM Scholarship application](https://user-images.githubusercontent.com/95768999/225399301-d350369f-990a-486a-b5cf-2b45c97af652.gif)

**Comment by barron-charles** [comment]
Thanks @joshkimux. Taking that and your previous subtask suggestion into account!

**Comment by joshkimux** [comment]
@barron-charles woohoo! Y'all are awesome.

**Comment by joshkimux** [comment]
@barron-charles great question! I haven't reviewed this in a while, but given its a list loop, most likely unspecific or repetitive button labels. This is a `defect-2` that can be resolved with `aria-label`s / `aria-labelledby` or one thing per page/subtask depending on your situation.

Y'all may want to recruit the help of a specialist via CAIA to support here as this can be a bit complicated to walkthrough.

**Comment by MichelleDieudonne** [comment]
Jira work item link: https://vajira.max.gov/browse/MBMS-37681

**Comment by sonntag-adam** [comment]
@coforma-terry @SarahKay8 fix is in staging.

**Comment by SarahKay8** [comment]
https://images.zenhubusercontent.com/133843125/5bd522ac-7075-4dcb-9246-65d6eb216b96/testing_.mp4 

Much more simpler design! Validated and closing @sonntag-adam @coforma-terry
